core/src/subgraph/context/instance.rs

```diff
@@ -37,6 +39,7 @@ where
         host_builder: T,
         host_metrics: Arc<HostMetrics>,
         offchain_monitor: &mut OffchainMonitor,
+        causality_region_next_value: CausalityRegion,
```

Comment:
from this I think I'd move the logic of next value to CausalityRegion, I think this next value is leaking the specifics of CR